test/e2e/performanceprofile/functests/13_llc/llc.go

```diff
@@ -191,6 +214,361 @@ var _ = Describe("[rfe_id:77446] LLC-aware cpu pinning", Label(string(label.Open
 			})
 		})
 	})
+	Context("Functional Tests", func() {
+		var L3CacheGroupSize int
+		BeforeAll(func() {
+			var (
+				numaCoreSiblings   map[int]map[int][]int
+				reserved, isolated []string
+				policy             = "single-numa-node"
+			)
+			profile, err := profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+			Expect(err).ToNot(HaveOccurred())
+			ctx := context.Background()
+			for _, cnfnode := range workerRTNodes {
+				numaInfo, err := nodes.GetNumaNodes(ctx, &cnfnode)
+				Expect(err).ToNot(HaveOccurred(), "Unable to get numa information from the node")
+				if len(numaInfo) < 2 {
+					Skip(fmt.Sprintf("This test need 2 Numa nodes. The number of numa nodes on node %s < 2", cnfnode.Name))
+				}
+
+				getCCX := nodes.GetL3SharedCPUs(&cnfnode)
+				cpus, err := getCCX(0)
+				Expect(err).ToNot(HaveOccurred())
+				L3CacheGroupSize = cpus.Size()
+				if len(numaInfo[0]) == L3CacheGroupSize {
+					Skip("This test requires systems where L3 cache is shared amount subset of cpus")
+				}
+			}
+			// Modify the profile such that we give 1 whole ccx to reserved cpus
+			By("Modifying the profile")
+			for _, node := range workerRTNodes {
+				numaCoreSiblings, err = nodes.GetCoreSiblings(ctx, &node)
+				Expect(err).ToNot(HaveOccurred())
+			}
+
+			// Get cpu siblings from core 0-7
+			// Assign one whole L3 Cache group for reserved cpus.
```

Comment:
this is a bit fragile but without `machineinfo` or similar is challenging to improve

Reply:
Agree, for now i have used cpuset.CPUSet . addressed in the latest commit